### PR TITLE
[ricos-editor] feat(callbacks): onToolbarButtonClick

### PR DESCRIPTION
**Pull request by Barackos**

### CODE REVIEW

File: packages/editor-common/web/src/Components/InlineToolbarButton.tsx [hunk 1]
```diff
@@ -70,6 +77,25 @@ class InlineToolbarButton extends Component<InlineToolbarButtonProps> {
...
+      pluginType,
+      dataHook,
+      isActive,
+      children,
+    } = this.props;
+    helpers?.onToolbarButtonClick?.({
+      buttonName: dataHook || '',
+      pluginId: pluginType,
+      version: Version.currentVersion,
+      value: !isLastAddStep || children ? undefined : String(!isActive),
```
Comment (lxgreen) [inline packages/editor-common/web/src/Components/InlineToolbarButton.tsx:94]: i don't understand this line. very confusing

File: packages/editor-common/web/src/Components/InlineToolbarButton.tsx [hunk 2]
```diff
@@ -20,11 +20,18 @@ type InlineToolbarButtonProps = {
   disabled?: boolean;
   buttonContent?: string;
   showArrowIcon?: boolean;
+  helpers?: Helpers;
+  isLastAddStep?: boolean;
+  pluginType?: string;
```
Comment (lxgreen) [inline packages/editor-common/web/src/Components/InlineToolbarButton.tsx:25]: the `InlineToolbarButton` is actually used outside, as a base for custom buttons in external toolbars (e.g. wix-data). The BI props are gonna complicate its API, even if they are optional. I suggest to move these props to a HOC and wrap the button, should be simple. Such HOC may come handy in the future for other components.

File: packages/editor-common/web/src/Components/LinkComponents/LinkButton.jsx [hunk 3]
```diff
@@ -3,16 +3,27 @@ import PropTypes from 'prop-types';
...
-  <InlineToolbarButton icon={icon || LinkIcon} dataHook={'LinkButton'} {...otherProps} />
+  <InlineToolbarButton
+    isLastAddStep={false}
+    icon={icon || LinkIcon}
+    dataHook={dataHook}
+    {...otherProps}
+  />
 );
 
+LinkButton.dataHook = dataHook;
```
Comment (lxgreen) [inline packages/editor-common/web/src/Components/LinkComponents/LinkButton.jsx:17]: what's this property for?

File: packages/editor-common/web/src/Components/LinkComponents/LinkButton.jsx [hunk 4]
```diff
@@ -3,16 +3,27 @@ import PropTypes from 'prop-types';
...
+
 LinkButton.propTypes = {
   onClick: PropTypes.func.isRequired,
   isActive: PropTypes.bool.isRequired,
   theme: PropTypes.object.isRequired,
+  helpers: PropTypes.object,
```
Comment (lxgreen) [inline packages/editor-common/web/src/Components/LinkComponents/LinkButton.jsx:23]: looks like this couple of props added to many components. HOC can help

File: packages/editor-common/web/src/Utils/draftUtils.ts [hunk 5]
```diff
@@ -819,6 +819,15 @@ export function isCursorAtStartOfContent(editorState: EditorState) {
   return isStartOfLine && isCursorAtFirstLine(editorState);
 }
 
+export const hasBlockType = (blockType: string, editorState: EditorState) => {
+  const currentBlockType = editorState
+    .getCurrentContent()
+    .getBlockForKey(editorState.getSelection().getStartKey())
+    .getType();
```
Comment (lxgreen) [inline packages/editor-common/web/src/Utils/draftUtils.ts:826]: hmm
we have tens of duplications of these lines. not sure what to do about it. we have to refactor the whole draft api usages across the whole codebase. we're lucky all this is gonna be removed this year.

File: packages/editor/web/src/RichContentEditor/Toolbars/buttons/utils/createTextBlockStyleButton.jsx [hunk 6]
```diff
@@ -93,14 +104,13 @@ export default ({ blockTypes, Icons, InactiveIcon = null, tooltipTextKey }) =>
...
       const dataHookText = `textBlockStyleButton_${textForHooks}`;
-
       return (
         <TextButton
           icon={Icon}
           theme={theme}
           isMobile={isMobile}
           isActive={this.blockTypeIsActive}
-          onClick={this.setBlockStyle}
+          onClick={event => this.setBlockStyle(event, textForHooks)}
```
Comment (lxgreen) [inline packages/editor/web/src/RichContentEditor/Toolbars/buttons/utils/createTextBlockStyleButton.jsx:113]: please move the function to const

File: packages/editor/web/src/RichContentEditor/Toolbars/buttons/utils/createTextAlignmentButton.jsx [hunk 7]
```diff
@@ -20,7 +21,12 @@ export default ({ alignment, Icon, tooltipTextKey }) =>
     isActive = () => this.props.alignment === alignment;
 
     handleClick = () => {
-      const { onClick, getEditorState, setEditorState } = this.props;
+      const { onClick, helpers, getEditorState, setEditorState } = this.props;
```
Comment (lxgreen) [inline packages/editor/web/src/RichContentEditor/Toolbars/buttons/utils/createTextAlignmentButton.jsx:24]: please consider HOC for all the cases where you modify the `onClick`.

**Review by lxgreen — CHANGES_REQUESTED** [review]

**Comment by lxgreen** [comment]
tldr -- try to HOC as much as possible to decrease duplication

File: packages/editor-common/web/src/Components/InlineToolbarButton.tsx (hunk 1 above)
Comment (Barackos) [reply packages/editor-common/web/src/Components/InlineToolbarButton.tsx]: removing this logic. turned out to be unnecessary
